ospatch/exec_step.go

```diff
@@ -0,0 +1,79 @@
+//  Copyright 2019 Google Inc. All Rights Reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package ospatch
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/GoogleCloudPlatform/guest-logging-go/logger"
+	osconfigpb "github.com/GoogleCloudPlatform/osconfig/_internal/gapi-cloud-osconfig-go/google.golang.org/genproto/googleapis/cloud/osconfig/v1alpha2"
+	"github.com/GoogleCloudPlatform/osconfig/common"
+)
+
+func getExecutablePath(ctx context.Context, stepConfig *osconfigpb.ExecStepConfig) (string, error) {
+	if gcsObject := stepConfig.GetGcsObject(); gcsObject != nil {
+		var reader io.ReadCloser
+		reader, err := common.FetchWithGCS(ctx, gcsObject.GetBucket(), gcsObject.GetObject(), gcsObject.GetGenerationNumber())
+		if err != nil {
+			return "", fmt.Errorf("error reading GCS object: %s", err)
+		}
+		defer reader.Close()
+		logger.Debugf("Fetched GCS object bucket %s object %s generation number %d", gcsObject.GetBucket(), gcsObject.GetObject(), gcsObject.GetGenerationNumber())
+
+		localPath := "./" + filepath.Base(gcsObject.GetObject())
```

Comment:
Use os.TempDir() here instead of ./, and path.Base as gcsObject.GetObject() is a uri

Also join with filepath.Join()

Reply:
filepath.Base(gcsObject.GetObject()) needs to be changed to path.Base(gcsObject.GetObject)

GCS objects are URIs so they are separated with '/' which is what path uses, filepath uses os.PathSeparator which is '\' on WIndows

Reply:
Ah understood, updated to path.Base, thanks